### PR TITLE
[Merged by Bors] - chore: adaptations for leanprover/lean4#3187

### PR DESCRIPTION
**Pull request by kim-em**
These are the adaptations for leanprover/lean4#3187 (new simp behaviour on ground terms), taken from the lean-pr-testing-3187 branch, which has now been merged into nightly-testing.

(Note this is a PR to bump/v4.6.0.)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: test/matrix.lean [hunk 1]
```diff
@@ -9,7 +9,7 @@ import Qq
 import Std.Tactic.GuardExpr
 
 open Qq
-
+set_option says.verify true
```
Comment (semorrison) [inline test/matrix.lean:12]:
```suggestion
```

File: test/matrix.lean [hunk 2]
```diff
@@ -9,7 +9,6 @@ import Qq
 import Std.Tactic.GuardExpr
 
 open Qq
```
Comment (semorrison) [inline test/matrix.lean:11]:
```suggestion
open Qq

```

File: Mathlib/RingTheory/Polynomial/Hermite/Basic.lean [hunk 3]
```diff
@@ -172,7 +172,7 @@ theorem coeff_hermite_explicit :
       simp only
       -- Factor out (-1)'s.
       rw [mul_comm (↑k + _ : ℤ), sub_eq_add_neg]
-      nth_rw 3 [neg_eq_neg_one_mul]
+      nth_rw 4 [neg_eq_neg_one_mul]
```
Comment (digama0) [inline Mathlib/RingTheory/Polynomial/Hermite/Basic.lean:175]: how concerning is this?

Comment (semorrison) [reply Mathlib/RingTheory/Polynomial/Hermite/Basic.lean:175]: It's wrong, and doesn't work, so I'm reverting it now, but there remains the mystery of why it has to be `4` and not `3` on nightly-testing!

Comment (semorrison) [reply Mathlib/RingTheory/Polynomial/Hermite/Basic.lean]: Ah, the proofs diverge later, and then one on nightly-testing is worse. Sorry, this was just noise from the complicated merges. I've cleanup up `nightly-testing` now so we won't get this diff showing up again.

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**